libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -131,34 +134,109 @@ export class ModelOverview extends React.Component<
         RegressionMetrics.MeanPrediction
       ];
     }
-    this.setState({
-      maxCohortId: this.getMaxCohortId(),
-      selectedDatasetCohorts: this.context.errorCohorts.map((errorCohort) => {
-        return errorCohort.cohort.getCohortID();
-      }),
-      selectedMetrics: defaultSelectedMetrics
-    });
+    this.setState(
+      {
+        selectedDatasetCohorts: this.context.errorCohorts.map((errorCohort) => {
+          return errorCohort.cohort.getCohortID();
+        }),
+        selectedMetrics: defaultSelectedMetrics
+      },
+      () =>
+        this.state.datasetCohortChartIsVisible
+          ? this.updateDatasetCohortStats()
+          : this.updateFeatureCohortStats()
+    );
+  }
+
+  private ifCohortIndexesEquals(a: number[], b: number[]): boolean {
+    return (
+      Array.isArray(a) &&
+      Array.isArray(b) &&
+      a.length === b.length &&
+      a.every((val, index) => val === b[index])
+    );
   }
 
   public componentDidUpdate(): void {
-    const maxCohortId = this.getMaxCohortId();
-    if (maxCohortId > this.state.maxCohortId) {
-      // A cohort has a higher ID than the previously recorded
-      // maximum which indicates that new cohorts were created.
-      const newCohorts = this.context.errorCohorts
-        .filter(
-          (errorCohort) =>
-            errorCohort.cohort.getCohortID() > this.state.maxCohortId
-        )
-        .map((errorCohort) => errorCohort.cohort.getCohortID());
-      this.setState({
-        maxCohortId,
-        selectedDatasetCohorts:
-          this.state.selectedDatasetCohorts?.concat(newCohorts)
-      });
+    const newDatasetCohortIDs = this.context.errorCohorts.map((errorCohort) => {
+      return errorCohort.cohort.getCohortID();
+    });
+    const oldDatasetCohortIDs = this.state.datasetBasedCohorts.map(
+      (errorCohort) => {
+        return errorCohort.cohort.getCohortID();
+      }
+    );
+    console.log(newDatasetCohortIDs);
+    console.log(oldDatasetCohortIDs);
+    if (!this.ifCohortIndexesEquals(newDatasetCohortIDs, oldDatasetCohortIDs)) {
+      let newOldDifference = newDatasetCohortIDs.filter(
```

Comment:
"removed items" and "added items" might be easier to understand.

Reply:
changed.